loadtest/user/userentity/user.go

```diff
@@ -78,13 +79,14 @@ func (ue *UserEntity) Connect() <-chan error {
 	}
 
 	go ue.listen(ue.wsErrorChan)
+	ue.connected = true
 	return ue.wsErrorChan
 }
 
 // Disconnect closes the websocket connection.
 func (ue *UserEntity) Disconnect() error {
 	cli := ue.getWsClient()
-	if cli == nil {
+	if cli == nil || !ue.connected {
```

Comment:
So I am wondering, now that we have this can we remove the mutex and the shared client object? As far as I can see `wsClient` can stay internal to the listen goroutine and synchronization from/to the outside done only through available channels.

Reply:
Yep !